code/ModelAdmin.php

```diff
@@ -183,6 +185,65 @@ public function Link($action = null)
         return parent::Link($action);
     }
 
+    /**
+     * Get the link for the tab of a DataObject subclass managed by this ModelAdmin.
+     *
+     * Note that if the modelClass is managed on multiple tabs and you want a link for a
+     * specific tab you should use {@link getLinkForModelTab} instead.
+     *
+     * @throws InvalidArgumentException if $modelClass is not managed by this ModelAdmin.
+     */
+    public function getLinkForModelClass(string $modelClass): string
+    {
+        if (!$this->isManagedModel($modelClass)) {
+            throw new InvalidArgumentException("$modelClass isn't managed by this ModelAdmin.");
+        }
+        return $this->getLinkForModelTab($this->getModelTabForModelClass($modelClass));
+    }
+
+    /**
+     * Get the link for a specific tab on this ModelAdmin.
+     *
+     * If you do not have multiple tabs for a given class you should use {@link getLinkForModelClass}
+     * instead.
+     *
+     * @param string $modelTab This is either the custom url segment for the tab or, if that
+     * was not not defined, it is the fully qualified class name of the managed model.
+     * In the latter case, you should prefer to use {@link getLinkForModelClass}.
+     *
+     * @throws InvalidArgumentException if $modelTab is not a tab on this ModelAdmin.
+     */
+    public function getLinkForModelTab(string $modelTab): string
+    {
+        // Don't use isManagedModel here because a subclass of a managed model may not have its own tab.
+        if (!array_key_exists($modelTab, $this->getManagedModels())) {
+            throw new InvalidArgumentException("$modelTab isn't a tab on this ModelAdmin.");
+        }
+        return $this->Link($this->sanitiseClassName($modelTab));
+    }
+
+    /**
+     * Get the link for editing an object inside this ModelAdmin.
+     *
+     * @throws InvalidArgumentException if $obj is not managed by this ModelAdmin.
+     */
+    public function getEditLinkForDataObject(DataObject $obj): string
```

Comment:
Steve and I have talked about this offline - I'm renaming this to `getEditLinkForManagedDataObject()`.
This is a little clearer here but it makes the method added in https://github.com/silverstripe/silverstripe-admin/pull/1361 a lot more comprehensible (I'll rename it to match).